client/request_handler.go

```diff
@@ -22,7 +30,16 @@ func NewRequestHandler(client BaseClient) *RequestHandler {
 	}
 }
 
-func (c *RequestHandler) sendRequest(method string, rawURL string, data url.Values,
+func NewRequestHandlerWithContext(client BaseClient) *RequestHandlerWithContext {
+	return &RequestHandlerWithContext{
+		Client: client,
+		Edge:   os.Getenv("TWILIO_EDGE"),
+		Region: os.Getenv("TWILIO_REGION"),
+		ctx:    context.TODO(),
+	}
+}
+
+func (c *RequestHandlerWithContext) sendRequest(method string, rawURL string, data url.Values,
```

Comment:
I think what you want here is to have the first method here be a context.Context and then you can edit the methods below to call this instead.